src/main/java/com/hltech/pact/gen/domain/pact/PactFactory.java

```diff
@@ -199,12 +202,18 @@ private static InteractionResponse createInteractionResponse(
         return InteractionResponse.builder()
                 .status(responseRepresentation.getStatus().toString())
                 .headers(mapHeaders(responseRepresentation.getHeaders()))
-                .body(BodySerializer.serializeBody(responseRepresentation.getBody(), objectMapper, podamFactory))
+                .body(buildBody(responseRepresentation, objectMapper))
                 .build();
     }
 
     private static Map<String, String> mapHeaders(List<Param> headers) {
         return headers.stream()
             .collect(Collectors.toMap(Param::getName, header -> String.valueOf(getParamValue(header))));
     }
+
+    private static JsonNode buildBody(ResponseRepresentation responseRepresentation, ObjectMapper objectMapper) {
+        return responseRepresentation.isEmptyBodyExpected()
+            ? JsonNodeFactory.instance.textNode(StringUtils.EMPTY)
```

Comment:
Why shouldn't we pass Void type of body or any other than add isEmptyBodyExpected()